packages/plugin-core/src/KeybindingUtils.ts

```diff
@@ -1,206 +1,322 @@
-        if (!_.isUndefined(entry.command)) {
-          return entry.command === "-extension.vim_navigateCtrlL";
-        } else {
-          return false;
-        }
-      }).length > 0;
 
-    if (alreadyHasOverride) {
-      return { data: { keybindingConfigPath } };
+    const userKeybindingConfig = readJSONWithCommentsSync(
+      keybindingConfigPath
+    ) as CommentJSONArray<Keybindings>;
+
+    const alreadyResolved: KeybindingConflict[] = [];
+
+    userKeybindingConfig.forEach((keybinding) => {
+      // find remap / disable of conflicting keybindings
+      let command = keybinding.command;
+      if (command.startsWith("-")) {
+        command = command.substring(1);
+      }
+
+      const resolvedConflict = conflicts.find(
+        (conflict) => conflict.commandId === command
+      );
+      if (resolvedConflict) {
+        alreadyResolved.push(resolvedConflict);
+      }
+
+      // find remap / disable of dendron keybindings
+      command = keybinding.command;
+      if (command.startsWith("-")) {
+        command = command.substring(1);
+      }
+      const resolvedDendronConflict = conflicts.find(
+        (conflict) => conflict.conflictsWith === command
+      );
+
+      if (resolvedDendronConflict) {
+        alreadyResolved.push(resolvedDendronConflict);
+      }
+    });
+
+    return _.differenceBy(conflicts, alreadyResolved);
+  }
+
+  static generateKeybindingBlockForCopy(opts: {
+    entry: Keybindings;
+    disable?: boolean;
+  }) {
+    const { entry, disable } = opts;
+    const whenClause = entry.when ? `  "when": "${entry.when}",` : undefined;
+
+    const args = entry.args
+      ? `  "args": ${JSON.stringify(entry.args)},`
+      : undefined;
+
+    const block = [
+      "{",
+      `  "key": "${disable ? entry.key : ""}",`,
+      `  "command": "${disable ? "-" : ""}${entry.command}",`,
+      whenClause,
+      args,
+      "}",
+      "",
+    ]
+      .filter((line) => line !== undefined)
+      .join("\n");
+    return block;
+  }
+
+  static async showKeybindingConflictPreview(opts: {
+    conflicts: KeybindingConflict[];
+  }) {
+    const md = _md();
+    const { conflicts } = opts;
+    const defaultKeybindingJSON =
+      await KeybindingUtils.openDefaultKeybindingFileAndGetJSON({
+        close: false,
+      });
+    if (defaultKeybindingJSON === undefined) {
+      throw new DendronError({
+        message: "Failed reading default keybinding.",
+        severity: ERROR_SEVERITY.MINOR,
+      });
     }
+    await KeybindingUtils.openGlobalKeybindingFileAndGetJSON({
+      close: false,
+    });
+    const keybindingJSONCommandUri = `command:workbench.action.openGlobalKeybindingsFile`;
+    const defaultKeybindingJSONCommandUri = `command:workbench.action.openDefaultKeybindingsFile`;
+    const contents = [
+      "# Known Keybinding Conflicts",
+      "",
+      "The keybindings listed below are known to have conflicts with default keybindings for Dendron commands.",
```

Comment:
nit: maybe instead of 'below', replace with 'at the bottom' - When I first read this my eyes wandered to the #### 1. / #### 2. section.